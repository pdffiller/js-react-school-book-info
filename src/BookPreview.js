import React from 'react';

const BookPreviewAuthor = ({ author }) => (
  <div className="book-preview--book_author">
    {author}
  </div>
);

const BookPreview = ({ thumbnail, title, authors, year }) => (
  <div className="book-preview">
    {thumbnail && <img src={thumbnail} alt={title} /> }
    <div className="book-preview--book_title">
      {title}
    </div>
    {
      authors && authors.map(
        author => <BookPreviewAuthor key={author} author={author} />
      )
    }
    <div className="book-preview--book_year">
      {year}
      {' '}
      {+year < 2000 ? 'old' : 'new'}
    </div>
  </div>
);

export default BookPreview;
