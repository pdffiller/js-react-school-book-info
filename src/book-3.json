{
  "kind": "books#volume",
  "id": "UOf3tgAACAAJ",
  "etag": "xqpPd5nfEpA",
  "selfLink": "https://www.googleapis.com/books/v1/volumes/UOf3tgAACAAJ",
  "volumeInfo": {
    "title": "Двенадцать стульев",
    "subtitle": "пьеса в 3-х действиях",
    "publisher": "Тип. но. 1 Госплана СССР",
    "publishedDate": "1960",
    "readingModes": {
      "text": false,
      "image": false
    },
    "pageCount": 4,
    "printedPageCount": 4,
    "dimensions": {
      "height": "23.00 cm"
    },
    "printType": "BOOK",
    "maturityRating": "NOT_MATURE",
    "allowAnonLogging": false,
    "contentVersion": "preview-1.0.0",
    "language": "ru",
    "previewLink": "http://books.google.com.ua/books?id=UOf3tgAACAAJ&hl=&source=gbs_api",
    "infoLink": "https://play.google.com/store/books/details?id=UOf3tgAACAAJ&source=gbs_api",
    "canonicalVolumeLink": "https://market.android.com/details?id=book-UOf3tgAACAAJ"
  },
  "saleInfo": {
    "country": "UA",
    "saleability": "NOT_FOR_SALE",
    "isEbook": false
  },
  "accessInfo": {
    "country": "UA",
    "viewability": "NO_PAGES",
    "embeddable": false,
    "publicDomain": false,
    "textToSpeechPermission": "ALLOWED",
    "epub": {
      "isAvailable": false
    },
    "pdf": {
      "isAvailable": false
    },
    "webReaderLink": "http://play.google.com/books/reader?id=UOf3tgAACAAJ&hl=&printsec=frontcover&source=gbs_api",
    "accessViewStatus": "NONE",
    "quoteSharingAllowed": false
  }
}