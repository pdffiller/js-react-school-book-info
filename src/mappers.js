export const getBookInfo = ({ id, volumeInfo }) => ({
  ...volumeInfo,
  id,
  year: +volumeInfo.publishedDate.slice(0, 4),
  thumbnail: volumeInfo.imageLinks && volumeInfo.imageLinks.thumbnail,
});
