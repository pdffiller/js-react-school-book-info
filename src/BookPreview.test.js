import React from 'react';
import renderer from 'react-test-renderer';
import BookPreview from './BookPreview';
import bookJSON from './book.json';
import { getBookInfo } from './mappers';


describe('BookPreview', () => {
  it('should be rendered without errors without props', () => {
    const html = renderer.create(<BookPreview />).toJSON();
    expect(html).toMatchSnapshot();
  });

  it('should be rendered without errors with a valid book info', () => {
    const book = getBookInfo(bookJSON);
    const html = renderer.create(
      <BookPreview {...book} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  it('should be rendered without errors with a old book info', () => {
    const book = getBookInfo(
      require('./book-3.json')
    );
    const html = renderer.create(
      <BookPreview {...book} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});