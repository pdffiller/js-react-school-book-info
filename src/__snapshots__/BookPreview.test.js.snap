// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookPreview should be rendered without errors with a old book info 1`] = `
<div
  className="book-preview"
>
  <div
    className="book-preview--book_title"
  >
    Двенадцать стульев
  </div>
  <div
    className="book-preview--book_year"
  >
    1960
     
    old
  </div>
</div>
`;

exports[`BookPreview should be rendered without errors with a valid book info 1`] = `
<div
  className="book-preview"
>
  <img
    alt="Happy"
    src="http://books.google.com/books/content?id=Ly0UAgAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&imgtk=AFLRE71XLkpkbKFOQhq6bhaIaC0aesSaSMp3yMGiGWXblwlos_ZvOSHqFXC716BvbNf-ltMU1N4-iuXwUyLvZ2N65g3MP1K7xIAe9qlUxKUntCNuORwYzA2-OPwb4C5F9jlpWhaOmc5C&source=gbs_api"
  />
  <div
    className="book-preview--book_title"
  >
    Happy
  </div>
  <div
    className="book-preview--book_author"
  >
    Marla Braziel
  </div>
  <div
    className="book-preview--book_year"
  >
    2013
     
    new
  </div>
</div>
`;

exports[`BookPreview should be rendered without errors without props 1`] = `
<div
  className="book-preview"
>
  <div
    className="book-preview--book_title"
  />
  <div
    className="book-preview--book_year"
  >
     
    new
  </div>
</div>
`;
